panic: send on closed channel on gc.

**What happened**:
https://github.com/juicedata/juicefs/actions/runs/12496360301/job/34868011495

```
2024/12/25 20:35:44.429856 juicefs[5631] <INFO>: Meta address: redis://127.0.0.1:6379/1 [NewClient@interface.go:511]
2024/12/25 20:35:44.430508 juicefs[5631] <WARNING>: AOF is not enabled, you may lose data if Redis is not shutdown properly. [checkRedisInfo@info.go:84]
2024/12/25 20:35:44.430608 juicefs[5631] <INFO>: Ping redis latency: 62.608µs [checkServerConfig@redis.go:3628]
2024/12/25 20:35:44.431094 juicefs[5631] <INFO>: Data use file:///var/jfs/myjfs/ [gc@gc.go:100]
2024/12/25 20:35:44.435219 juicefs[5631] <INFO>: using 2 slices (87124664 bytes) [gc@gc.go:273]
2024/12/25 20:35:44.443154 juicefs[5631] <INFO>: scanned 1204 objects, 0 valid, 0 pending delete (0 bytes), 0 compacted (0 bytes), 0 leaked (0 bytes), 0 delslices (0 bytes), 0 delfiles (0 bytes), 1204 skipped (3027087753 bytes) [gc@gc.go:393]
2024/12/25 20:35:44.443206 juicefs[5631] <INFO>: flush session 0: [FlushSession@base.go:619]
2024/12/25 20:35:44.443227 juicefs[5631] <INFO>: close session 0: <nil> [CloseSession@base.go:608]
warning: GOCOVERDIR not set, no coverage data emitted
2024/12/25 20:35:44.470976 juicefs[5639] <INFO>: Meta address: redis://127.0.0.1:6379/1 [NewClient@interface.go:511]
2024/12/25 20:35:44.471639 juicefs[5639] <WARNING>: AOF is not enabled, you may lose data if Redis is not shutdown properly. [checkRedisInfo@info.go:84]
2024/12/25 20:35:44.471746 juicefs[5639] <INFO>: Ping redis latency: 55.525µs [checkServerConfig@redis.go:3628]
2024/12/25 20:35:44.472093 juicefs[5639] <INFO>: Data use file:///var/jfs/myjfs/ [gc@gc.go:100]
2024/12/25 20:35:44.495657 juicefs[5639] <INFO>: using 2 slices (871[2466](https://github.com/juicedata/juicefs/actions/runs/12496360301/job/34868011495#step:5:2467)4 bytes) [gc@gc.go:273]
panic: send on closed channel

goroutine 70 [running]:
github.com/juicedata/juicefs/cmd.gc.func1({0xc000cdd400, 0x2, 0x3e8?})
	/home/runner/work/juicefs/juicefs/cmd/gc.go:132 +0x16a
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).newMsg(0xc0001b2008, 0x3e8, {0xc000cdd400, 0x2, 0x2})
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:390 +0x106
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).deleteSlice_(0xc0001b2008, 0x147, 0x48b202)
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:2334 +0xf9
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).NewSession.func1()
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:489 +0x68
created by github.com/juicedata/juicefs/pkg/meta.(*baseMeta).NewSession in goroutine 1
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:487 +0x74b
test_gc_trash_slices exit with 2
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/cmd/gc.go
+++ b/cmd/gc.go
@@ -29,6 +29,7 @@ import (
 	osync "github.com/juicedata/juicefs/pkg/sync"
 	"github.com/juicedata/juicefs/pkg/utils"
 	"github.com/juicedata/juicefs/pkg/vfs"
+	"github.com/pkg/errors"
 
 	"github.com/urfave/cli/v2"
 )
@@ -122,27 +123,13 @@ func gc(ctx *cli.Context) error {
 
 	var wg sync.WaitGroup
 	var delSpin *utils.Bar
-	var sliceChan chan meta.Slice // pending delete slices
 
 	if delete || compact {
 		delSpin = progress.AddCountSpinner("Cleaned pending slices")
-		sliceChan = make(chan meta.Slice, 10240)
 		m.OnMsg(meta.DeleteSlice, func(args ...interface{}) error {
 			delSpin.Increment()
-			sliceChan <- meta.Slice{Id: args[0].(uint64), Size: args[1].(uint32)}
-			return nil
+			return store.Remove(args[0].(uint64), int(args[1].(uint32)))
 		})
-		for i := 0; i < threads; i++ {
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				for s := range sliceChan {
-					if err := store.Remove(s.Id, int(s.Size)); err != nil {
-						logger.Warnf("remove %d_%d: %s", s.Id, s.Size, err)
-					}
-				}
-			}()
-		}
 	}
 
 	c := meta.WrapContext(ctx.Context)
@@ -367,11 +354,8 @@ func gc(ctx *cli.Context) error {
 		}
 	}
 	m.OnMsg(meta.DeleteSlice, func(args ...interface{}) error {
-		return nil
+		return errors.New("stop deleting slice")
 	})
-	if sliceChan != nil {
-		close(sliceChan)
-	}
 	close(leakedObj)
 	wg.Wait()
 	if delete || compact {

